activesupport/lib/active_support/notifications/fanout.rb

```diff
@@ -91,11 +91,14 @@ def self.new(pattern, listener, monotonic)
           if listener.respond_to?(:start) && listener.respond_to?(:finish)
             subscriber_class = Evented
           else
-            # Doing all this to detect a block like `proc { |x| }` vs
-            # `proc { |*x| }` or `proc { |**x| }`
-            if listener.respond_to?(:parameters)
-              params = listener.parameters
-              if params.length == 1 && params.first.first == :opt
+            # Doing all this to detect a single argument block or callable
+            # like `proc { |x| }` vs `proc { |*x| }`, `proc { |**x| }`,
+            # or `proc { |x, **y| }`
+            if listener.respond_to?(:call)
```

Comment:
`listener` has to respond to `call`, since we're always going to call `call` on it later. I don't think we need this check.

Reply:
Yup, good ***call*** 😏